File: business-backend/app/schemas/product.py
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


# 基础商品模型
class ProductBase(BaseModel):
    name: str
    description: Optional[str] = None
    credits: int
    price: float
    image_url: Optional[str] = None
    is_active: bool = True
    sort_order: int = 0


# 创建商品时的数据模型
class ProductCreate(ProductBase):
    pass


# 更新商品时的数据模型
class ProductUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    credits: Optional[int] = None
    price: Optional[float] = None
    image_url: Optional[str] = None
    is_active: Optional[bool] = None
    sort_order: Optional[int] = None


# 返回给API的商品模型
class Product(ProductBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True 

File: admin-frontend/src/types/product.ts
// 商品类型定义
export interface Product {
  id: number
  name: string
  description: string
  credits: number
  price: number
  image_url: string
  is_active: boolean
  sort_order: number
  created_at: string
  updated_at: string
}

// 创建商品请求参数
export interface CreateProductRequest {
  name: string
  description: string
  credits: number
  price: number
  image_url: string
  is_active: boolean
  sort_order: number
}

// 更新商品请求参数
export interface UpdateProductRequest {
  name?: string
  description?: string
  credits?: number
  price?: number
  image_url?: string
  is_active?: boolean
  sort_order?: number
}

// 商品列表查询参数
export interface ProductQueryParams {
  page?: number
  limit?: number
  is_active?: boolean
  search?: string
}

// 分页响应结果
export interface PaginatedResponse<T> {
  items: T[]
  total: number
  page: number
  per_page: number
  total_pages: number
}
